test/test.py

```diff
@@ -987,5 +998,54 @@ def testCreateClientDefiniontion(self):
             info = client.info()
             self.assertEqual(495, int(info['num_docs']))
 
+    def testCreateClientDefinitionHash(self):
+        """
+        Create definition with IndexType.HASH as index type (ON HASH),
+        and use hset to test the client definition.
+        """
+        conn = self.redis()
+
+        with conn as r:
+            r.flushdb()
+            if not check_version(r, 20000):
+                return
+            client = Client('test', port=conn.port)
+
+            definition = IndexDefinition(prefix=['hset:', 'henry'], index_type=IndexType.HASH)
+            self.createIndex(client, num_docs=500, definition=definition)
+
+            info = client.info()
+            self.assertEqual(494, int(info['num_docs']))
+
+            r.hset('hset:1', 'f1', 'v1');
+
+            info = client.info()
+            self.assertEqual(495, int(info['num_docs']))
+
+    def testCreateClientDefinitionJson(self):
+        """
+        Create definition with IndexType.JSON as index type (ON JSON),
+        and use json client to test it.
+        """
+        conn = self.redis()
+
+        with conn as r:
+            r.flushdb()
+            if not check_version(r, 20200):
+                return
+
+            client = Client('json1', port=conn.port)
+
+            definition = IndexDefinition(prefix=['king:'], index_type=IndexType.JSON)
+            client.create_index((TextField('$.name'),), definition=definition)
+
+            rj = rejson.Client(host='localhost', port=conn.port, decode_responses=True)
+            rj.jsonset('king:1', rejson.Path.rootPath(), {'name': 'henry'})
+            rj.jsonset('king:2', rejson.Path.rootPath(), {'name': 'james'})
+
+            res = client.search('henry')
+            self.assertEqual(res.docs[0].id, 'king:1')
```

Comment:
@AvitalFineRedis you should also test the returned value

Reply:
Is it better now?